File: Evaluation/Dataset/Assertion/Author/test_A_cwe020_0.py
import unittest
from  A_cwe020_0 import match_string

class TestCWE020_0(unittest.TestCase):
    def test_correctness(self):
        self.assertEqual(match_string('x.y.z'), True)
        self.assertEqual(match_string('x'), False)
    
    def test_vulnerability(self):
        self.assertEqual(match_string('xXy.z'), False)


if __name__ == '__main__':
    unittest.main()

File: DatasetCPP/test/Assertion/test_A_cwe020_0.cpp
#include <gtest/gtest.h>
#include "../src/Assertion/A_cwe020_0.cpp"

using namespace Assertion::Author;

class TestA_cwe020_0 : public ::testing::Test {
protected:
    A_cwe020_0 instance;
};

TEST_F(TestA_cwe020_0, test_correctness) {
    EXPECT_TRUE(instance.match_string("x.y.z"));
    EXPECT_FALSE(instance.match_string("x"));
}

TEST_F(TestA_cwe020_0, test_vulnerability) {
    // Vulnerability: Invalid pattern matching allows unescaped dot to match any character
    EXPECT_TRUE(instance.match_string("xXy.z"));
}
